doc/modules/classes.rst

```diff
@@ -760,47 +760,92 @@ Plotting
 
 .. currentmodule:: sklearn
 
+Linear classifiers
+------------------
+.. autosummary::
+   :toctree: generated/
+   :template: class.rst
+
+   linear_model.LogisticRegression
+   linear_model.LogisticRegressionCV
+   linear_model.PassiveAggressiveClassifier
+   linear_model.Perceptron
+   linear_model.RidgeClassifier
+   linear_model.RidgeClassifierCV
+   linear_model.SGDClassifier
+
+Classical linear regressors
+---------------------------
+
 .. autosummary::
    :toctree: generated/
    :template: class.rst
 
-   linear_model.ARDRegression
-   linear_model.BayesianRidge
    linear_model.ElasticNet
    linear_model.ElasticNetCV
-   linear_model.HuberRegressor
-   linear_model.Lars
-   linear_model.LarsCV
    linear_model.Lasso
    linear_model.LassoCV
+   linear_model.LinearRegression
+   linear_model.Ridge
+   linear_model.RidgeCV
+   linear_model.SGDRegressor
+
+Bayesian regressors
+-------------------
+
+.. autosummary::
+   :toctree: generated/
+   :template: class.rst
+
+   linear_model.ARDRegression
+   linear_model.BayesianRidge
+
+Regressors with variable selection
+----------------------------------
+
+.. autosummary::
+   :toctree: generated/
+   :template: class.rst
+
+   linear_model.Lars
+   linear_model.LarsCV
    linear_model.LassoLars
    linear_model.LassoLarsCV
    linear_model.LassoLarsIC
-   linear_model.LinearRegression
-   linear_model.LogisticRegression
-   linear_model.LogisticRegressionCV
-   linear_model.MultiTaskLasso
-   linear_model.MultiTaskElasticNet
-   linear_model.MultiTaskLassoCV
-   linear_model.MultiTaskElasticNetCV
    linear_model.OrthogonalMatchingPursuit
    linear_model.OrthogonalMatchingPursuitCV
-   linear_model.PassiveAggressiveClassifier
-   linear_model.PassiveAggressiveRegressor
-   linear_model.Perceptron
+
+Outlier-robust regressors
+-------------------------
+
+.. autosummary::
+   :toctree: generated/
+   :template: class.rst
+
+   linear_model.HuberRegressor
    linear_model.RANSACRegressor
-   linear_model.Ridge
-   linear_model.RidgeClassifier
-   linear_model.RidgeClassifierCV
-   linear_model.RidgeCV
-   linear_model.SGDClassifier
-   linear_model.SGDRegressor
    linear_model.TheilSenRegressor
 
+Multi-task linear regressors
+----------------------------
+
+.. autosummary::
+   :toctree: generated/
+   :template: class.rst
+
+   linear_model.MultiTaskElasticNet
+   linear_model.MultiTaskElasticNetCV
+   linear_model.MultiTaskLasso
+   linear_model.MultiTaskLassoCV
```

Comment:
"Multi-task" in this context is a combination of multi-output and shared sparsity support (variable selection). I think we should add a sentence to explain this because it could mean many other things in different contexts.